gtsam/base/base.i

```diff
@@ -81,13 +81,6 @@ class IndexPairSetMap {
   gtsam::IndexPairSet at(gtsam::IndexPair& key);
 };
 
-class DSFMapIndexPair {
```

Comment:
thanks for the PR, Varun. Can you explain this change a bit more? We use this code in GTSFM as:
```python
        # Generate the DSF to form tracks
        dsf = gtsam.DSFMapIndexPair()
        track_2d_list = []
        # for DSF finally
        # measurement_idxs represented by ks
        for (i1, i2), k_pairs in matches_dict.items():
            for (k1, k2) in k_pairs:
                dsf.merge(gtsam.IndexPair(i1, k1), gtsam.IndexPair(i2, k2))

        key_set = dsf.sets()

        erroneous_track_count = 0
        # create a landmark map: a list of tracks
        # Each track is represented as a list of (camera_idx, measurements)
        for set_id in key_set:
            index_pair_set = key_set[set_id]  # key_set is a wrapped C++ map, so this unusual syntax is required

            # Initialize track from measurements
            track_measurements = []
            for index_pair in gtsam.IndexPairSetAsArray(index_pair_set):
```

Reply:
so i believe this would be a breaking change. I'm guessing the new syntax would be cleaner, but I'm just wondering what changes we would need to make.

Reply:
This should not break anything since I simply uncommented the templated version of this function a few lines above. The syntax in the wrapper should be identical.

Reply:
Maybe just the name will be different? @johnwlambert Could you test if this is the case or not?

Reply:
Ah I see @varunagrawal, I had forgotten that the wrapper will concatenate the template type to the end of the name, so the name should be unchanged.

To make sure the functionality is the same as before, I expanded some unit tests in this other PR: https://github.com/borglab/gtsam/pull/854/files

If #854 looks ok to you, maybe we can merge it into develop, then pull develop into this branch, and as long as all the tests still pass, feel free to merge.

Reply:
Done and done.